pkg/render/whisker/component.go

```diff
@@ -0,0 +1,427 @@
+
+	GuardianContainerName              = "guardian"
+	GoldmaneContainerName              = "goldmane"
+	WhiskerContainerName               = "whisker"
+	WhiskerBackendContainerName        = "whisker-backend"
+	GuardianServiceName                = "tigera-guardian"
+	ManagedClusterConnectionSecretName = "tigera-managed-cluster-connection"
+)
+
+func Whisker(cfg *Configuration) render.Component {
+	c := &Component{cfg: cfg}
+
+	tolerations := append(c.cfg.Installation.ControlPlaneTolerations, rmeta.TolerateCriticalAddonsAndControlPlane...)
+	if c.cfg.Installation.KubernetesProvider.IsGKE() {
+		tolerations = append(tolerations, rmeta.TolerateGKEARM64NoSchedule)
+	}
+
+	return c
+}
+
+// Configuration contains all the config information needed to render the component.
+type Configuration struct {
+	PullSecrets                 []*corev1.Secret
+	OpenShift                   bool
+	Installation                *operatorv1.InstallationSpec
+	TunnelSecret                *corev1.Secret
+	TrustedCertBundle           certificatemanagement.TrustedBundle
+	LinseedPublicCASecret       *corev1.Secret
+	ManagementClusterConnection *operatorv1.ManagementClusterConnection
+}
+
+type Component struct {
+	cfg *Configuration
+
+	goldmaneImage       string
+	guardianImage       string
+	whiskerImage        string
+	whiskerBackendImage string
+}
+
+func (c *Component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.cfg.Installation.Registry
+	path := c.cfg.Installation.ImagePath
+	prefix := c.cfg.Installation.ImagePrefix
+
+	var err error
+
+	c.whiskerImage, err = components.GetReference(components.ComponentCalicoWhisker, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	c.whiskerBackendImage, err = components.GetReference(components.ComponentCalicoWhiskerBackend, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	c.goldmaneImage, err = components.GetReference(components.ComponentCalicoGoldmane, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	c.guardianImage, err = components.GetReference(components.ComponentCalicoGuardian, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *Component) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		render.CreateNamespace(WhiskerNamespace, c.cfg.Installation.KubernetesProvider, render.PSSRestricted, c.cfg.Installation.Azure),
+	}
+
+	objs = append(objs,
+		render.CreateOperatorSecretsRoleBinding(WhiskerNamespace),
+		c.serviceAccount(),
+		c.clusterRole(),
+		c.clusterRoleBinding(),
+		c.goldmaneService(),
+		c.whiskerService(),
+		c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
+
+	if c.cfg.ManagementClusterConnection != nil {
+		objs = append(objs,
+			secret.CopyToNamespace(WhiskerNamespace, c.cfg.TunnelSecret)[0],
+			// TODO maybe the controller needs to create this?
+			c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace),
```

Comment:
We need to write another controller to handle this, per my other comment. Right now, both core_controller and whisker controller are going to be writing the exact same configmap but with different contents.

Reply:
bundle is removed from this PR.